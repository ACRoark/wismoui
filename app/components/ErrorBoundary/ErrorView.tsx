import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import ColumnLayout from 'components/ColumnLayout';
import useConfig from 'hooks/useConfig';
import ExclamationPoint from 'images/ExclamationPoint';

import messages from './messages';

import './index.less';

interface IErrorViewProps {
  errorInfo: Error | null;
}

const ErrorView: React.FC<IErrorViewProps> = (props: IErrorViewProps): React.ReactElement => {
  const { errorInfo } = props;
  const { urls } = useConfig();

  const onButtonClick = (): void => {
    window.location.href = urls.backToMIMO;
  };

  return (
    <ColumnLayout>
      <div className="dte-wismo-error-boundary-error-view">
        <div className="banner">
          <div>
            <ExclamationPoint className="error-icon" title="error" />
          </div>
          <div className="error-message">
            <FormattedMessage {...messages.errorMessage} />
            {process.env.NODE_ENV !== 'production' && <div className="error-details">{errorInfo?.message}</div>}
          </div>
        </div>
        <Button className="button" dataTrackSubAction="no orders go to core" type="primary" onClick={onButtonClick}>
          <FormattedMessage {...messages.backToMimoButtonText} />
        </Button>
      </div>
    </ColumnLayout>
  );
};

export default ErrorView;
