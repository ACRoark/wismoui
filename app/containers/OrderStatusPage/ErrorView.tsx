import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import ColumnLayout from 'components/ColumnLayout';
import useConfig from 'hooks/useConfig';
import ExclamationPoint from 'images/ExclamationPoint';

import messages from './messages';

import './index.less';

interface IErrorViewProps {
  reason: string;
}

const ErrorView: React.FC<IErrorViewProps> = (props: IErrorViewProps): React.ReactElement => {
  const { reason } = props;

  const { urls } = useConfig();

  const onButtonClick = (): void => {
    window.location.href = urls.backToMIMO;
  };

  return (
    <ColumnLayout>
      <div className="dte-wismo-order-status-page-error-view">
        <div className="banner">
          <div>
            <ExclamationPoint className="error-icon" title="error" />
          </div>
          <div className="error-message">{reason}</div>
        </div>
        <Button className="button" dataTrackSubAction="no orders go to core" type="primary" onClick={onButtonClick}>
          <FormattedMessage {...messages.backToMimoButtonText} />
        </Button>
      </div>
    </ColumnLayout>
  );
};

export default ErrorView;
