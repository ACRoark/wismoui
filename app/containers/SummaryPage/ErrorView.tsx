import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import ColumnLayout from 'components/ColumnLayout';
import useConfig from 'hooks/useConfig';
import ExclamationPoint from 'images/ExclamationPoint';

import messages from './messages';

import './index.less';

const ErrorView: React.FC = (): React.ReactElement => {
  const { urls } = useConfig();

  const onButtonClick = (): void => {
    window.location.href = urls.backToMIMO;
  };

  return (
    <ColumnLayout>
      <div className="dte-wismo-summary-page-error-view">
        <div className="banner">
          <div>
            <ExclamationPoint className="error-icon" title="error" />
          </div>
          <div className="error-message">
            <FormattedMessage {...messages.system_unavailable_message} />
          </div>
        </div>
        <Button className="button" dataTrackSubAction="no orders go to core" type="primary" onClick={onButtonClick}>
          <FormattedMessage {...messages.backToMimoButtonText} />
        </Button>
      </div>
    </ColumnLayout>
  );
};

export default ErrorView;
