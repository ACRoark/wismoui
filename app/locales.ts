export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
