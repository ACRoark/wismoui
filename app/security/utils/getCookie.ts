const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
