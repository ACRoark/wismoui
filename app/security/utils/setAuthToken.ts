import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
