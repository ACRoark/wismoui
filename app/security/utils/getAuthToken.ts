import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
