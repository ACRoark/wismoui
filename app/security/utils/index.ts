import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
