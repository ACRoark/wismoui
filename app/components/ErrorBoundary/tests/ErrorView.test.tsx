import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import ErrorView from '../ErrorView';

const store = createMockStore();

describe('<ErrorView />', (): void => {
  const error = new Error('I am an error');
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<Provider store={store}><ErrorView errorInfo={error}/></Provider>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should render correctly when given error info in ${language}`, (): void => {
      const tree = createSnapshotWithIntl(<Provider store={store}><ErrorView errorInfo={error}/></Provider>, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
