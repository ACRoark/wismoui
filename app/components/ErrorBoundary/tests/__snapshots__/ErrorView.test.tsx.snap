// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorView /> should render correctly when given error info in en 1`] = `
<div
  className="dte-wismo-column-layout"
>
  <div
    className="dte-wismo-error-boundary-error-view"
  >
    <div
      className="banner"
    >
      <div>
        <svg
          className="error-icon"
          fill="none"
          height="23"
          viewBox="0 0 6 23"
          width="6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            error
          </title>
          <path
            d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="error-message"
      >
        An error has occurred and we apologize for the inconvenience. Please go back and try again.
        <div
          className="error-details"
        >
          I am an error
        </div>
      </div>
    </div>
    <button
      className="button dte-wismo-button dte-wismo-button-primary"
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      data-track-sub-action="no orders go to core"
      onClick={[Function]}
    >
      Go to Start or Stop Service
    </button>
  </div>
</div>
`;

exports[`<ErrorView /> should render correctly when given error info in es 1`] = `
<div
  className="dte-wismo-column-layout"
>
  <div
    className="dte-wismo-error-boundary-error-view"
  >
    <div
      className="banner"
    >
      <div>
        <svg
          className="error-icon"
          fill="none"
          height="23"
          viewBox="0 0 6 23"
          width="6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            error
          </title>
          <path
            d="M4.43457 15.4961H1.88574L1.12402 0.583984H5.18164L4.43457 15.4961ZM0.948242 20.1689C0.948242 19.4463 1.13867 18.8994 1.51953 18.5283C1.90039 18.1475 2.44238 17.957 3.14551 17.957C3.83887 17.957 4.37109 18.1523 4.74219 18.543C5.11328 18.9336 5.29883 19.4756 5.29883 20.1689C5.29883 20.8623 5.1084 21.4141 4.72754 21.8242C4.35645 22.2246 3.8291 22.4248 3.14551 22.4248C2.45215 22.4248 1.91016 22.2295 1.51953 21.8389C1.13867 21.4482 0.948242 20.8916 0.948242 20.1689Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="error-message"
      >
        Se produjo un error y le pedimos disculpas por las molestias. Por favor, regrese y vuelva a intentarlo.
        <div
          className="error-details"
        >
          I am an error
        </div>
      </div>
    </div>
    <button
      className="button dte-wismo-button dte-wismo-button-primary"
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      data-track-sub-action="no orders go to core"
      onClick={[Function]}
    >
      Ir a Iniciar o Detener el Servicio
    </button>
  </div>
</div>
`;
